python/ml4ir/base/model/scoring/monte_carlo_scorer.py

```diff
@@ -99,12 +88,21 @@ def __init__(
                          logs_dir=logs_dir,
                          **kwargs)
 
-        self.monte_carlo_test_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS][MonteCarloInferenceKey.NUM_TEST_TRIALS]
-        self.monte_carlo_training_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS][MonteCarloInferenceKey.NUM_TRAINING_TRIALS]
+        self.use_fixed_mask_in_training = bool(self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS].get(
+            MonteCarloInferenceKey.USE_FIXED_MASK_IN_TRAINING, False))
+        QueryFeatureMask.masking_config[MonteCarloInferenceKey.USE_FIXED_MASK_IN_TRAINING] = self.use_fixed_mask_in_training
+        if not self.use_fixed_mask_in_training:
+            self.monte_carlo_training_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS][
+                MonteCarloInferenceKey.NUM_TRAINING_TRIALS]
+            QueryFeatureMask.masking_config[MonteCarloInferenceKey.NUM_TRAINING_TRIALS] = self.monte_carlo_training_trials
 
-        # Adding 1 here to account of the extra inference run with training=False.
-        self.monte_carlo_test_trials_tf = tf.constant(self.monte_carlo_test_trials + 1, dtype=tf.float32)
-        self.monte_carlo_training_trials_tf = tf.constant(self.monte_carlo_training_trials + 1, dtype=tf.float32)
+        self.use_fixed_mask_in_testing = bool(self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS].get(
+            MonteCarloInferenceKey.USE_FIXED_MASK_IN_TESTING, False))
+        QueryFeatureMask.masking_config[MonteCarloInferenceKey.USE_FIXED_MASK_IN_TESTING] = self.use_fixed_mask_in_testing
+        if not self.use_fixed_mask_in_testing:
+            self.monte_carlo_test_trials = self.model_config[MonteCarloInferenceKey.MONTE_CARLO_TRIALS][
+                MonteCarloInferenceKey.NUM_TEST_TRIALS]
+            QueryFeatureMask.masking_config[MonteCarloInferenceKey.NUM_TEST_TRIALS] = self.monte_carlo_test_trials
```

Comment:
@mohazahran can you move these default values out as well? Easier to read and track in some config file.

Reply:
do you mean rely on config to set the default values?